crypto/fipsmodule/sha/sha3.c

```diff
@@ -109,49 +109,43 @@ uint8_t *SHAKE256(const uint8_t *data, const size_t in_len, uint8_t *out, size_t
   return out;
 }
 
-int SHAKE_Init(KECCAK1600_CTX *ctx, size_t block_size) {
-  // The SHAKE block size depends on the security level of the algorithm only
-  // It is independent of the output size
-  ctx->block_size = block_size;
-  return SHA3_Init(ctx, SHAKE_PAD_CHAR, 0);
+// FIPS202 APIs manage internal input/output buffer on top of Keccak1600 API layer
+static void FIPS202_Reset(KECCAK1600_CTX *ctx) {
+  memset(ctx->A, 0, sizeof(ctx->A));
```

Comment:
Should this also zero out `ctx->buf`?
```suggestion
  OPENSSL_memset(ctx->A, 0, sizeof(ctx->A));
```

Reply:
The `buf` is only accessible through the `buf_load` value. So, even if it is not zero'ed, it should not be an issue. Once that data is placed into the buffer, the `buf_load` is assigned the number of bytes that are 'useful' and accessible. When `buf_load` is 0, there is no useful data in the buffer.
However, I can zero it.

Reply:
No need to zero.